submissions/dashakim/oop-task/js/app.js

```diff
@@ -0,0 +1,106 @@
+var gameScore = 0;
+
+const ENEMY = {
+  speed: 100,
+  varianceSpeed: 222,
+  location: [70, 155, 235]
+};
+const PLAYER = {
+  startX: 200,
+  startY: 404,
+  stepX: 101,
+  stepY: 83
+};
+
+const CANVAS = {
+  begin: -60,
+  end: 510,
+  leftSide: 0,
+  rightSide: 400,
+  top: 0,
+  bottom: 404,
+  halfCell: 50
+};
+
+var Draw = function() {};
+
+Draw.prototype.render = function() {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+var Enemy = function(x, y, speed, player) {
+  this.x = x;
+  this.y = y;
+  this.speed = speed;
+  this.player = player;
+  this.sprite = "images/car.png";
+};
+
+Enemy.prototype = Object.create(Draw.prototype);
+Enemy.prototype.constructor = Enemy;
+
+Enemy.prototype.update = function(dt) {
+  this.x += this.speed * dt;
+  if (this.x > CANVAS.end) {
+    this.x = CANVAS.begin;
+    //random speed for enemies
+    this.speed = ENEMY.speed + Math.floor(Math.random() * ENEMY.varianceSpeed);
+  }
+  // return to start position player when collide with enemy
+  if (
+    player.x > this.x - CANVAS.halfCell &&
+    player.x < this.x + CANVAS.halfCell &&
+    player.y > this.y - CANVAS.halfCell &&
+    player.y < this.y + CANVAS.halfCell
+  ) {
+    player.startPosition();
+  }
+};
+
+var Player = function() {
+  this.startPosition();
+  this.sprite = "images/char-cat-girl.png";
+};
+Player.prototype = Object.create(Draw.prototype);
+Player.prototype.constructor = Player;
+Player.prototype.update = function() {
+  if (this.y <= 0) {
+    displayStats();
```

Comment:
An illegal guest from the outer scope. Pass this method at `Player` instantiation and keep it among `Player` class props.

How to pass and store a function:
```
var SomeClass = function (callback) {
 this.callback = callback;
 // .................
 callback();
}

function doSomething() {
  // ...
}

var instance1 = new SomeClass(doSomething()); // wrong; this will pass the result of doSomething call
var instance2 = new SomeClass(doSomething); // correct; this will pass the function per se
```

Reply:
I changed the scope for "score", but I must confess, I didn't understand if I need to put method or something else  into variable like in this example:
var instance2 = new SomeClass(doSomething); // correct; this will pass the function per se

Reply:
You can pass a method/function per se as an argument to use/call it later. Once passed as an argument such function is called a [callback function](https://codeburst.io/javascript-what-the-heck-is-a-callback-aba4da2deced). Callbacks are used in real world tasks very often.

In this case you definitely need to pass a method/function.

Reply:
Yes, I'm understand callback functions in general, but I didn't use in real case.  I hope I did it right for now